src/utils/xmlParser.ts

```diff
@@ -38,27 +39,16 @@ export async function parseXML(
 	const saxWasmBuffer = await initSaxWasm();
 	const saxParser = new SAXParser(SaxEventType.CloseTag + SaxEventType.OpenTag);
 
-	saxParser.eventHandler = parseHandler;
-
 	// Instantiate and prepare the wasm for parsing
 	if (!await saxParser.prepareWasm(saxWasmBuffer)) {
 		throw new Error("Unknown error during WASM Initialization");
 	}
 
-	// stream from a file in the current directory
-	contentStream.on("data", (chunk: Uint8Array) => {
-		try {
-			saxParser.write(chunk);
-		} catch (err) {
-			if (err instanceof Error) {
-				// In case of an error, destroy the content stream to make the
-				// error bubble up to our callers
-				contentStream.destroy(err);
-			} else {
-				throw err;
-			}
-		}
-	});
+	const webContentStream = Readable.toWeb(contentStream);
+
+	for await (const [event, detail] of saxParser.parse(webContentStream.getReader())) {
+		parseHandler(event, detail);
+	}
+
 	await finished(contentStream);
```

Comment:
is finished still required? I would expect that the iterator is exhaustive? The sax-wasm README does not call finished in the ReadableStream example...

BTW: toWeb is still experimental in Node 23 (according to their docs)

Reply:
Yeah I guess not but I thought it shouldn't hurt leaving it there. But good point about the experimental API. Then we shouldn't use it, yet.